apps/reference/docs/guides/auth/auth-security.mdx

```diff
@@ -0,0 +1,503 @@
+standard JWT claims for this purpose. At this time each JWT is signed using the
+`HS256` algorithm (HMAC with SHA256). The secret key used to derive the
+signature is generated from a cryptographically-secure pseudo random number
+generator when a project is initially created and is 512 bits long. This key is
+stored encrypted in our database, so Supabase staff are unable to readily
+access it.
+
+We believe in giving project owners maximum flexibility, so this key is visible
+to you only in your project's dashboard. Make sure you never share this value
+with parties you don't completely trust.
+
+:::tip
+You can always rotate the JWT signing key if you suspect it has leaked.
+Rotating keys on a regular basis is a good practice. Note that rotating a key
+means that all access tokens will immediately become invalid. Make sure you do
+not forget to update all other services that you use Supabase Auth access
+tokens in with the rotated secret to avoid any disruptions.
+:::
+
+Access tokens are also short lived, and you can configure their validity
+duration in the project's settings page. The default value at this time is 1
+hour, which is common across the industry. We do not recommend exceeding this
+duration period, and recommend using a shorter value if it makes sense for your
+application. Durations less than 5 minutes are discouraged as computers often
+may have significant clock skew that could introduce confusing and difficult to
+debug application errors and behaviors. Shortening the duration also increases
+the application's dependence of the uptime of the authentication server.
+
+The purpose of access tokens is to be readily shared between components of your
+application, be it the browser, Supabase services such as PostgREST and
+functions, your hosting or other API servers under your control. Due to
+their short lifetime, any security incident involving them will have a
+time-limited blast radius.
+
+Always make sure that access tokens are passed as headers or cookies, and avoid
+passing them as URL query parameters (regardless of request methods) to prevent
+them from appearing in access logs which may not be under your total control.
+When using them in Supabase functions or other API servers, take care to not
+accidentally write them to logs.
+
+It is for this reason that Supabase Auth returns both the access token and
+refresh token to your browser application via the URL fragment (anything after
+the `#` sign), as browsers will actually avoid sending this data to the
+underlying server. Thus this makes sure that the access and refresh token are
+received only by the JavaScript running in your browser and does not end up in
+access logs or is visible by your application's hosting service. Even if the
+servers hosting your browser application are under your total control, we've
+found that often there are various other appliances that intercept HTTP
+requests where disabling logging is difficult or sometimes impossible.
```

Comment:
```suggestion
Supabase Auth returns the access token and
refresh token to your browser application via the URL fragment (anything after
the `#` sign). This prevents browsers sending the data to the
server. URL fragments are only
accessible to the JavaScript running in your browser.
```

I removed a chunk from this one as it's been repeated a few times and it feels like this is a big vague: "e've
found that often there are various other appliances that intercept HTTP
requests where disabling logging is difficult or sometimes impossibl"